### PR TITLE
Enhancements to disk layout docs.

### PR DESCRIPTION
**Pull request by igor-stoppa**
Add more explanations about the dsk format and how to customize it.
Also, fix typos and formatting.

Signed-off-by: Igor Stoppa <igor.stoppa@intel.com>

### CODE REVIEW

**Comment by igor-stoppa** [comment]
@dbkinder some addition to the docs

**Comment by kad** [comment]
[skip ci]

File: doc/architecture/disk-layout.rst [hunk 1]
```diff
@@ -22,6 +22,11 @@ The minimum number of partitions is 3:
   atomic update of the EFI application (with type 0x2700)
 - Main root filesystem, by default ext4 (type 0x8300)
 
+The first two EFI partitions are necessary to ensure the
+device will still retain its capability to boot, even in
```
Comment (dbkinder) [inline doc/architecture/disk-layout.rst]: grammar police would want this to say, "even in **the** presence.."

Comment (dbkinder) [inline doc/architecture/disk-layout.rst]: should be, "even in **the** presence of..."

Comment (igor-stoppa) [inline doc/architecture/disk-layout.rst]: done

**Comment by mythi** [comment]
+1

**Pull request merged**